server/models/board.py

```diff
@@ -0,0 +1,127 @@
+"""
+'  Class modeling a box game board
+'
+'  rows:  number of rows in the grid
+'  cols:  number of columns in the grid
+'
+'  edges are maintained in a map of maps with the following structure:
+'    { sourcePoint1: {neighboringPoint1: owner, neighboringPoint2: owner}, sourcePoint2: ... }
+'
+'    where each point is a row-majored tuple of 2 integers describing the row and column of the point.
+'
+'  completed boxes are maintained in a map mapping the top-left corner of the box to the owner of the box
+"""
+
+
+class Board(object):
+
+    def __init__(self, rows, cols):
+        self.rows = rows
+        self.cols = cols
+        self.edges = {}  # maps edges to edge owner (board[p1][p2] = Owner)
+        self.boxes = {}  # maps top-left point of each box to box owner
+
+        self._reset_board()
+
+    '''
+    '' Resets the board and box maps
+    ''
+    '' Adds unowned edges to each possible position on the board
+    '''
+    def _reset_board(self):
+        self.boxes = {}
+        for r in range(0, self.rows):
+            for c in range(0, self.cols):
+
+                if not (r, c) in self.edges:
+                    self.edges[(r, c)] = {}
+
+                # add an un-owned edge from the current node to the node below and to the right
+                # do not add an edge that goes out of bounds (r + 1 exceeds rows, c + 1 exceeds cols)
+                if not c == self.cols - 1:
+                    self.edges[(r, c)][(r, c + 1)] = None
+
+                if not r == self.rows - 1:
+                    self.edges[(r, c)][(r + 1, c)] = None
+
+    '''
+    '' Claims an unowned edge from p1 to p2
+    ''
+    ''  p1:     A tuple describing source node (row, col)
+    ''  p2:     A tuple describing target node (row, col)
+    ''  player: the owner of the new edge
+    ''
+    ''  This implementation coerces the source node to always be above or to the left of the destination node
+    ''    eg: if claim_edge((1, 1), (0, 1), PLAYER) is called, board[(0, 1)][(1, 1)] is assigned to PLAYER
+    ''
+    ''  Raises an ValueError if the edge is already owned
+    ''  Raises a ValueError if the edge isn't between two adjacent nodes
+    '''
+    def claim_edge(self, p1, p2, owner):
```

Comment:
Yeah, it's undirected.  The point of coercing is to make sure the same edge owner is returned when calling
`claim_edge((0,0),(0,1))` and `claim_edge((1,0),(0,0))`.  Since they refer to the same edge, they need to be ordered correctly in the edge map